src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -164,29 +170,33 @@ def deploy(
         raise ClickException(msg)
 
     # Create stage
-    stage_name = snowpark.stage_name
-    stage_manager = StageManager()
-    stage_name = FQN.from_string(stage_name).using_context()
-    stage_manager.create(fqn=stage_name, comment="deployments managed by Snowflake CLI")
-
     snowflake_dependencies = _read_snowflake_requrements_file(
         paths.snowflake_requirements_file
     )
-
-    artifact_stage_directory = get_app_stage_path(stage_name, snowpark.project_name)
-    artifact_stage_target = (
-        f"{artifact_stage_directory}/{paths.artifact_file.path.name}"
+    stage_names = {
+        entity.stage for entity in [*functions.values(), *procedures.values()]
+    }
+    stage_manager = StageManager()
+    project_name = (
+        pd.defaults.project_name if pd.defaults.project_name else "my_snowpark_project"
```

Comment:
Should we put the default "my_snowpark_project" value under `SnowparkEntity`?
Also, each defaults entry will automatically apply to each entity that has this field but was not specified by the user. Not sure this check is needed here.

Reply:
+1

Reply:
Done